### PR TITLE
BUG: SparseArray.astype(np.int64)

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/sparse/array.py [hunk 1]
```diff
@@ -1305,24 +1307,16 @@ def astype(self, dtype: AstypeArg | None = None, copy: bool = True):
...
             # GH#34457
-            if isinstance(future_dtype, np.dtype):
-                values = np.array(self)
-                return astype_nansafe(values, dtype=future_dtype)
-            else:
-                # pylint: disable-next=used-before-assignment
-                dtype = cast(ExtensionDtype, dtype)
-                cls = dtype.construct_array_type()
-                return cls._from_sequence(self, dtype=dtype, copy=copy)
+            values = np.asarray(self)
```
Comment (mroeschke) [inline pandas/core/arrays/sparse/array.py:1310]: Is this robust if the original type was `pd.SparseDtype(pd.Int64Dtype)` with an `NA` and `astype(pd.Int32Dtype)` was called?

Comment (mroeschke) [reply pandas/core/arrays/sparse/array.py:1310]: Oh looks like there an existing bug here too (missing import)

```
In [1]: arr = pd.array([1, pd.NA], dtype=pd.SparseDtype(pd.Int64Dtype()))

In [2]: arr
Out[2]:
[1, <NA>]
Fill: <NA>
IntIndex
Indices: array([0], dtype=int32)

In [3]: arr.dtype
Out[3]: Sparse[Int64, <NA>]

In [4]: arr.astype(pd.Int16Dtype())
NameError: name 'ExtensionDtype' is not defined
```

Comment (mroeschke) [reply pandas/core/arrays/sparse/array.py:1310]: Could you raise a `NotImplementedError` for the EADtypes here for now then?

Comment (jbrockmendel) [reply pandas/core/arrays/sparse/array.py:1310]: id rather do that as separately to Do It Right along with a couple other bugs mentioned in #49631

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**